core/templates/dev/head/services/ExplorationFeaturesService.ts

```diff
@@ -24,17 +24,21 @@ import { downgradeInjectable } from '@angular/upgrade/static';
   providedIn: 'root'
 })
 export class ExplorationFeaturesService {
+  static serviceIsInitialized = false;
   static settings = {
     isImprovementsTabEnabled: false,
     isPlaythroughRecordingEnabled: false,
-    areParametersEnabled: false
+    areParametersEnabled: false,
   }
 
   // TODO(#7176): Replace 'any' with the exact type. This has been kept as
   // 'any' because 'explorationData' and 'featuresData' are dicts with
   // underscore_cased keys which give tslint errors against underscore_casing
   // in favor of camelCasing.
   init(explorationData: any, featuresData: any): void {
+    if (ExplorationFeaturesService.serviceIsInitialized) {
```

Comment:
I'm not sure how this service is initialized when switching tabs, as I only see this being initialized in the `/exploration-editor-page.controller.ts`, is there any other place which intializes this service?

Reply:
No it's only initialized there, I just added this to be safe than sorry later.

Reply:
I see, thanks for the explanation! :)